src/howl/core.cljc

```diff
@@ -211,8 +211,8 @@
     PREDICATE  = PREFIXED_NAME / WRAPPED_IRI / ABSOLUTE_IRI / LABEL
     OBJECT     = BLANK_NODE / PREFIXED_NAME / WRAPPED_IRI / ABSOLUTE_IRI / LABEL
     DATATYPE   = PREFIXED_NAME / WRAPPED_IRI / ABSOLUTE_IRI / LABEL
-    LITERAL    = #'.+(?=@(\\w|-)+)' LANG /
-                 #'.+(?=\\^\\^\\S+)' '^^' DATATYPE /
+    LITERAL    = CHAR+ LANG /
+                 CHAR+ <'^^'> DATATYPE /
```

Comment:
This is the right approach. I want to keep stuff like whitespace in the parse, so we can make minor changes without messing with the user's preferred formatting. So I might want to keep the `^^`, but it's an edge case and I'm not sure.